### ISSUE TITLE
Bug: `no-unnecessary-type-constraint` missing placeholder in suggestion message

### ISSUE THREAD

**Issue by bmish** [typescript-eslint/typescript-eslint#5460]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.7.2&sourceType=module&code=GYVwdgxgLglg9mABAEwIZVQHgCqIKYAeUeYyAzoqmAJ4B8AFAJSIDeiAvkA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0ipWmQ5Nk+aEOb50URNGgdokcGAC+IXUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
function data<T extends any>() { }
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-unnecessary-type-constraint": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Offers suggestion:

```
Remove the unnecessary `any` constraint.
```

### Actual Result

Offers suggestion:

```
Remove the unnecessary `{{constraint}}` constraint.
```

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.33.0` |
| `@typescript-eslint/parser`            | `5.33.0` |
| `TypeScript`                           | `4.7.4` |
| `ESLint`                               | `8.21.0` |
| `node`                                 | `18.170` |
